logicalplan/plan.go

```diff
@@ -46,6 +46,9 @@ func New(expr parser.Expr, opts *query.Options) Plan {
 	setOffsetForAtModifier(opts.Start.UnixMilli(), expr)
 	setOffsetForInnerSubqueries(expr, opts)
 
+	// the engine handles sorting at the presentation layer
+	expr = trimSorts(expr)
```

Comment:
How do we support sort_desc right now?
The presentation layer needs to know it needs to sort in reverse order?

Reply:
And I wonder the behavior of topk and bottomk together with this change

Reply:
Sorting is done here https://github.com/thanos-io/promql-engine/blob/7a183df0925bee2179aa351b235ed090b82dca9e/engine/engine.go#L475. Care needs to be taken to compute the sorting order before this step in the plan but if that is done then sorting functions are irrelevant for the remaining computation.

Reply:
Thanks. Then this makes sense